magento/magento2
# #7627 Cookie expiration time is not extended when browsing the frontend

Current pull request fixes specified issue by extending Magento_PageCache module functionality.

### Description
The issue is related to enabled FPC and both its modes: "Built-in Cache" & "Varnish Cache".
Implemented different solution for each caching application:
* Extended front controller build-in plugin for "Built-in Cache" mode.
* Added a JS widget to send post ajax request to prolong cookie action for "Varnish Cache" mode, because the request may be processed by varnish completely.

The following things were done:
* Added cookie prolongation JS widget with corresponding block, template & layout update.
* Added cookie prolongation interface with corresponding model for frontend cookie.
* Added prolong cookie controller action.
* Updated front controller build-in plugin.
* Added tests for implemented functionality.

### Fixed Issues (if relevant)
1. magento/magento2#7627: Cookie expiration time is not extended when browsing the frontend

### Manual testing scenarios
Scenario 1:
1. Check "Page Cache" cache type is enabled in Backend.
Follow: "System" / "Cache Management"
2. Check "Build-in Cache" caching application for "Page Cache" cache type is selected in Backend.
Follow: "Stores" / "Configuration" / "Advanced" / "System" / "Full Page Cache"
3. Clean cache.
4. Browse the Frontend.
5. Check cookie expiration time is prolonged on every page.

Scenario 2:
1. Check "Page Cache" cache type is enabled in Backend:
Follow: "System" / "Cache Management"
2. Check "Varnish Cache" caching application for "Page Cache" cache type is selected in Backend.
Follow: "Stores" / "Configuration" / "Advanced" / "System" / "Full Page Cache"
3. Clean cache.
4. Browse the Frontend.
5. Check cookie expiration time is prolonged on every page.


### Contribution checklist
